File: modules/core/src/encoder/segment/SegmentEncoderMap.ts
import {

  CoreTCEncoder,
  PublisherTCEncoder,
  OOBVendorsEncoder,

} from '.';

export class SegmentEncoderMap {

  public readonly core: typeof CoreTCEncoder = CoreTCEncoder
  public readonly vendorsDisclosed: typeof OOBVendorsEncoder = OOBVendorsEncoder;
  public readonly vendorsAllowed: typeof OOBVendorsEncoder = OOBVendorsEncoder;
  public readonly publisherTC: typeof PublisherTCEncoder = PublisherTCEncoder;

}

File: GDPR-Phase2-Java/src/main/java/com/gdpr/phase2/encoder/segment/SegmentEncoderMap.java
package com.gdpr.phase2.encoder.segment;

import lombok.*;

@Getter @Setter
public class SegmentEncoderMap {
    private SegmentEncoderMap() {
    }
    private static final SegmentEncoderMap instance = new SegmentEncoderMap();
    public static SegmentEncoderMap getInstance() {
        return instance;
    }

    public final CoreTCEncoder core = CoreTCEncoder.getInstance();
    public final OOBVendorsEncoder vendorsDisclosed = OOBVendorsEncoder.getInstance();
    public final OOBVendorsEncoder vendorsAllowed = OOBVendorsEncoder.getInstance();
    public final PublisherTCEncoder publisherTC = PublisherTCEncoder.getInstance();

}
